packages/cli/src/lib/live-reload.ts

```diff
@@ -0,0 +1,88 @@
+// Support Remix 0.19 HMR and HDR.
+// Requirements:
+// - Enable `future.v2_dev` in `remix.config.js`
+// - Add <LiveReload /> to the root of your app
+// Note:
+// - This is not a public API in Remix and may change at any time.
+
+import http from 'node:http';
+import type {AssetsManifest} from '@remix-run/dev';
+import type {Result as RemixBuildResult} from '@remix-run/dev/dist/result.js';
+import type {Context as RemixContext} from '@remix-run/dev/dist/compiler/context.js';
+
+type LiveReloadState = {
+  manifest?: AssetsManifest;
+  prevManifest?: AssetsManifest;
+  loaderChanges?: Promise<RemixBuildResult<Record<string, string>>>;
+  prevLoaderHashes?: Record<string, string>;
+};
+
+export async function setupLiveReload(devServerPort: number) {
+  try {
+    const [{updates: hmrUpdates}, {serve}, {detectLoaderChanges}, {ok, err}] =
+      await Promise.all([
+        import('@remix-run/dev/dist/devServer_unstable/hmr.js'),
+        import('@remix-run/dev/dist/devServer_unstable/socket.js'),
+        import('@remix-run/dev/dist/devServer_unstable/hdr.js'),
+        import('@remix-run/dev/dist/result.js'),
```

Comment:
Are these other dependencies from Remix that we are hoping have unofficially stabilized?

Reply:
Yeah, we need to revive the conversation with them around these internal apis.